core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -0,0 +1,417 @@
+      LoadTableResponse response = loadInternal(ident);
+      String fullName = fullTableName(ident);
+
+      String tablePath = tablePath(ident);
+      Pair<RESTClient, FileIO> clients = tableClients(response.config());
+      TableMetadata base = response.tableMetadata();
+
+      Map<String, String> properties = propertiesBuilder.build();
+      TableMetadata replacement = base.buildReplacement(
+          schema,
+          spec != null ? spec : PartitionSpec.unpartitioned(),
+          writeOrder != null ? writeOrder : SortOrder.unsorted(),
+          location != null ? location : base.location(),
+          properties);
+
+      ImmutableList.Builder<MetadataUpdate> changes = ImmutableList.builder();
+
+      if (replacement.changes().stream().noneMatch(MetadataUpdate.SetCurrentSchema.class::isInstance)) {
+        // ensure there is a change to set the current schema
+        changes.add(new MetadataUpdate.SetCurrentSchema(replacement.currentSchemaId()));
+      }
+
+      if (replacement.changes().stream().noneMatch(MetadataUpdate.SetDefaultPartitionSpec.class::isInstance)) {
+        // ensure there is a change to set the default spec
+        changes.add(new MetadataUpdate.SetDefaultPartitionSpec(replacement.defaultSpecId()));
+      }
+
+      if (replacement.changes().stream().noneMatch(MetadataUpdate.SetDefaultSortOrder.class::isInstance)) {
+        // ensure there is a change to set the default sort order
+        changes.add(new MetadataUpdate.SetDefaultSortOrder(replacement.defaultSortOrderId()));
+      }
+
+      RESTTableOperations ops = new RESTTableOperations(
+          clients.first(), tablePath, clients.second(),
+          RESTTableOperations.UpdateType.REPLACE, changes.build(), base);
+
+      return Transactions.replaceTableTransaction(fullName, ops, replacement);
+    }
+
+    @Override
+    public Transaction createOrReplaceTransaction() {
+      // return a create or a replace transaction, depending on whether the table exists
+      // deciding whether to create or replace can't be determined on the service because schema field IDs are assigned
+      // at this point and then used in data and metadata files. because create and replace will assign different
+      // field IDs, they must be determined before any writes occur
+      try {
+        return replaceTransaction();
+      } catch (NoSuchTableException e) {
+        return createTransaction();
+      }
+    }
+
+    private LoadTableResponse stageCreate() {
+      String ns = RESTUtil.urlEncode(ident.namespace());
+      Map<String, String> properties = propertiesBuilder.build();
+
+      CreateTableRequest request = CreateTableRequest.builder()
+          .withName(ident.name())
+          .withSchema(schema)
+          .withPartitionSpec(spec)
+          .withWriteOrder(writeOrder)
+          .withLocation(location)
+          .setProperties(properties)
+          .build();
+
+      // TODO: will this be a specific route or a modified create?
```

Comment:
Right now, this uses a new route to state a table create. I haven't added this to the REST spec yet because it is an open question whether we want to use the existing create route with a query param or request flag, or use a separate route like this. Any opinions?

Reply:
I personally vote for query param.  It's almost an entire duplication of the create route if we go with a separate endpoint.